### PR TITLE
docs(@angular-devkit/schematics): add Task to glossary table

### LINKED ISSUE #22764: @angular-devkit/schematics glossary is missing Task
The [Schematics Glossary](https://github.com/angular/angular-cli/tree/master/packages/angular_devkit/schematics#glossary) doesn't introduce what a `Task` is, making it difficult for new schematic developers to know what `Task`s should be used for and how they work.

The _Schematics for libraries_ page on angular.io [demonstrates an example for adding a task](https://angular.io/guide/schematics-for-libraries#providing-installation-support), but it also doesn't provide an explanation on what a `Task` is. When do they run (before/after the `Tree` has been transformed by `Rule`s)? Do they run in parallel? How do you manage `Task` dependencies?

This issue is to document `Task` in the Schematics Glossary table so that it's easier for developers to create their own `Task`s.

### PR DESCRIPTION
**Pull request by ArthurDibe**
Include explanation about Task
Update glossary with explanation about Task
Fixes #22764

## PR Checklist

Please check to confirm your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular-cli/blob/main/CONTRIBUTING.md#-commit-message-guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?
Task is not listed within the glossary table for the documentation of @angular-devkit/schematics

Issue Number: 22764

## What is the new behavior?
Task is included within the glossary table for the documentation of @angular-devkit/schematics

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information
Updated documentation for @angular-devkit/schematics

### CODE REVIEW

**Comment by ArthurDibe** [comment]
Please let me know if there are any changes that I'm missing with this PR. Thanks in advance 😊

File: packages/angular_devkit/schematics/README.md [hunk 1]
```diff
@@ -20,6 +20,7 @@ What distinguishes Schematics from other generators, such as Yeoman or Yarn Crea
 | **Source**     | A function that creates an entirely new `Tree` from an empty filesystem. For example, a file source could read files from disk and create a Create Action for each of those. |
 | **Action**     | An atomic operation to be validated and committed to a filesystem or a `Tree`. Actions are created by schematics.                                                            |
 | **Sink**       | The final destination of all `Action`s.                                                                                                                                      |
+| **Task**       | A Task is a way to execute an external command or script in a schematic. A Task can be used to perform actions such as installing dependencies, running tests, or building a project. A Task is created by using the SchematicContext object and can be scheduled to run before or after the schematic tree is applied. |
```
Comment (alan-agius4) [inline packages/angular_devkit/schematics/README.md:23]:
```suggestion
| **Task**       | A Task is a way to execute an external command or script in a schematic. A Task can be used to perform actions such as installing dependencies, running tests, or building a project. A Task is created by using the `SchematicContext` object and can be scheduled to run before or after the schematic tree is applied. |
```

**Comment by alan-agius4** [comment]
Can you please format the changed doc using `yarn ng-dev format changed` and squash all the commits?

**Review by alan-agius4 — APPROVED** [review]
LGTM, thanks for the contribution.

**Pull request merged**